providers/os/resources/packages/windows_packages.go

```diff
@@ -203,15 +214,90 @@ func (w *WinPkgManager) Format() string {
 	return "win"
 }
 
-const installedAppsScript = `
-Get-ItemProperty (@(
-  'HKLM:\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*',
-  'HKCU:\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*',
-  'HKLM:\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*',
-  'HKCU:\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*'
-) | Where-Object { Test-Path $_ }) |
-Select-Object -Property DisplayName,DisplayVersion,Publisher,EstimatedSize,InstallSource,UninstallString | ConvertTo-Json -Compress
-`
+func (w *WinPkgManager) getLocalInstalledApps() ([]Package, error) {
+	pkgs := []string{
+		"HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall",
+		"HKCU\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall",
+		"HKLM\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall",
+		"HKCU\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall",
+	}
+	packages := []Package{}
+	for _, r := range pkgs {
+		// we get all the packages, found under the pkgs paths
+		children, err := registry.GetNativeRegistryKeyChildren(r)
+		if err != nil {
+			continue
+		}
+		for _, c := range children {
+			// for each package the information is contained as children of that registry's key
+			p := w.getPackageFromRegistryKey(c)
+			if p == nil {
+				continue
+			}
+			packages = append(packages, *p)
+		}
+	}
+	return packages, nil
+}
+
+func (w *WinPkgManager) getInstalledApps() ([]Package, error) {
+	if w.conn.Type() == shared.Type_Local {
+		return w.getLocalInstalledApps()
+	}
+
+	cmd, err := w.conn.RunCommand(powershell.Wrap(installedAppsScript))
```

Comment:
Just for my own knowledge, whey we switched the use of `Encode()` to `Wrap()` here?

Reply:
i will revert, unintentional